src/components/modalPrompt/ModalPromptReducers.ts

```diff
@@ -0,0 +1,72 @@
+import * as _ from 'underscore';
+import { IModalPromptActionPayload, ModalPromptAction } from './ModalPromptActions';
+import { IReduxAction } from '../../utils/ReduxUtils';
+
+export interface IModalPromptState {
+  id: string;
+  isOpened: boolean;
+  confirmed: boolean;
+}
+
+export const modalPromptInitialState: IModalPromptState = { id: undefined, isOpened: false, confirmed: false };
+export const modalPromptsInitialState: IModalPromptState[] = [];
+
+export const modalPromptReducer = (state: IModalPromptState = modalPromptInitialState,
+  action: IReduxAction<IModalPromptActionPayload>): IModalPromptState => {
+  switch (action.type) {
+    case ModalPromptAction.addModalPrompt:
+      return {
+        id: action.payload.id,
+        isOpened: false,
+        confirmed: false
```

Comment:
in those actions you should probably extends the state and override only the properties that you want to edit, e.g., in `cancelModalPrompt`

```
return state.id !== action.payload.id
  ? state
  : {
    ...state
    isOpened: false
  }
```